addons/pos_loyalty/models/loyalty_reward.py

```diff
@@ -11,3 +11,8 @@ def _get_discount_product_values(self):
         for vals in res:
             vals.update({'taxes_id': False})
         return res
+
+    def unlink(self):
+        if len(self) == 1 and self.env['pos.order.line'].search_count([('reward_id', 'in', self.ids)], limit=1):
```

Comment:
don't forget to add sudo() : self.env['pos.order.line'].sudo().search_count([('reward_id', 'in', self.ids)], limit=1)
because if a user have not access or in multi company env.
@tde-banana-odoo

Reply:
Indeed, thanks for the notice.
(and please stop pinging Thibault on scopes he's not working on, I believe he has enough notifications already).